crates/shared/src/trade_finding/external.rs

```diff
@@ -232,8 +236,7 @@ mod dto {
         pub executed_amount: U256,
         pub receiver: H160,
         pub valid_to: u32,
-        #[serde_as(as = "BytesHex")]
-        pub app_data: Vec<u8>,
```

Comment:
What's the reason for this change? The original version is more correct, no?

Reply:
The reason is that the AppDataHash needs to be constructed later here:
https://github.com/cowprotocol/services/blob/0617f4f92d9d2dd9d6c3a160fa239d1fc6824f63/crates/shared/src/price_estimation/trade_verifier.rs#L429
Which means that the Vec<u8> needs to be encoded back to String and only then to the AppDataHash. The change avoids back and forth decoding/encoding of the original String from the response.

Reply:
True. How about using the `AppDataHash` directly since it already implements a `Deserializer`:
[`9d23043` (#3038)](https://github.com/cowprotocol/services/pull/3038/commits/9d23043cddc515c964d68f77f718cce0fa8996d2)